packages/react-reconciler/src/__tests__/ReactHooksWithNoopRenderer-test.internal.js

```diff
@@ -1144,6 +1144,90 @@ function loadModules({
               ]);
             });
           });
+
+          it('does not warn about state updates for unmounted components with pending passive unmounts', () => {
+            let completePendingRequest = null;
+            function Component() {
+              Scheduler.unstable_yieldValue('Component');
+              const [didLoad, setDidLoad] = React.useState(false);
+              React.useLayoutEffect(() => {
+                Scheduler.unstable_yieldValue('layout create');
+                return () => {
+                  Scheduler.unstable_yieldValue('layout destroy');
+                };
+              }, []);
+              React.useEffect(() => {
+                Scheduler.unstable_yieldValue('passive create');
+                // Mimic an XHR request with a complete handler that updates state.
+                completePendingRequest = () => setDidLoad(true);
+                return () => {
+                  Scheduler.unstable_yieldValue('passive destroy');
+                };
+              }, []);
+              return didLoad;
+            }
+
+            act(() => {
+              ReactNoop.renderToRootWithID(<Component />, 'root', () =>
+                Scheduler.unstable_yieldValue('Sync effect'),
+              );
+              expect(Scheduler).toFlushAndYieldThrough([
+                'Component',
+                'layout create',
+                'Sync effect',
+              ]);
+              ReactNoop.flushPassiveEffects();
+              expect(Scheduler).toHaveYielded(['passive create']);
+
+              // Unmount but don't process pending passive destroy function
+              ReactNoop.unmountRootWithID('root');
+              expect(Scheduler).toFlushAndYieldThrough(['layout destroy']);
+
+              // Simulate an XHR completing, which will cause a state update-
+              // but should not log a warning.
+              completePendingRequest();
+
+              ReactNoop.flushPassiveEffects();
+              expect(Scheduler).toHaveYielded(['passive destroy']);
+            });
+          });
+
+          it('still warns about state updates for unmounted components with no pending passive unmounts', () => {
```

Comment:
Can you add a test for when there's a pending unmount on a different component?

Reply:
Sure!